src/radical/pilot/agent/agent_0.py

```diff
@@ -703,5 +713,63 @@ def _prepare_env(self, env_name, env_spec):
         return out
 
 
+    # --------------------------------------------------------------------------
+    #
+    def _start_service_ep(self):
+
+        if self._cfg.enable_ep:
+
+            self._service = ru.zmq.Server(uid='%s.server' % self._uid)
+            self._service.register_request('submit_tasks', self._ep_submit_tasks)
+            self._service.start()
+
+            self._log.info('service_url : %s', self._service.addr)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _ep_submit_tasks(self, request):
+
+      # import pprint
+      # self._log.debug('service request: %s', pprint.pformat(request))
+
+        tasks = request['tasks']
+
+        for task in tasks:
+
+            td = task['description']
+
+            if not task.get('uid'):
+                task['uid'] = ru.generate_id('task.ep.%(item_counter)04d',
+                                             ru.ID_CUSTOM, ns=self._pid)
+
+            sbox = '%s/%s' % (os.environ['RP_PILOT_SANDBOX'], td['uid'])
+
+            task['state']             = rps.AGENT_STAGING_INPUT_PENDING
+            task['task_sandbox_path'] = sbox
+            task['task_sandbox']      = 'file://localhost/' + sbox
+            task['pilot_sandbox']     = os.environ['RP_PILOT_SANDBOX']
+            task['session_sandbox']   = os.environ['RP_SESSION_SANDBOX']
+            task['resource_sandbox']  = os.environ['RP_RESOURCE_SANDBOX']
+            task['pilot']             = os.environ['RP_PILOT_ID']
```

Comment:
the last comment here:

should these attributes be set the same way as for service tasks? -> env variable wouldn't be available for the agent, right?
https://github.com/radical-cybertools/radical.pilot/blob/3366307dffbd2d1994de9d7f78c462ef43795eb8/src/radical/pilot/agent/agent_0.py#L333-L344

Reply:
Ack, I made that code consistent now.